kedro/io/data_catalog.py

```diff
@@ -754,7 +754,7 @@ def shallow_copy(
             dataset_patterns = self._sort_patterns(unsorted_dataset_patterns)
         else:
             dataset_patterns = self._dataset_patterns
-        return DataCatalog(
+        return self.__class__(
```

Comment:
```suggestion
        return cls(
```

How about using `cls` directly? It seems a bit weird to use `self.__class__` in this case. This is also how we define the factory method `from_config`.


https://github.com/kedro-org/kedro/blob/b6e585f80ace1896e6bd3c31d827a1991ed2c664/kedro/io/data_catalog.py#L327-L333

Reply:
The `shallow_copy()` method is supposed to copy an _instance_ of `DataCatalog`, so I don't think it makes sense to make it a `classmethod`. You'd then also lose the ability to reference any of the class arguments like `self._dataset_patterns`.

Reply:
I see an alternative using `type` directly, but in this case maybe `self.__class__` is more clear

https://stackoverflow.com/questions/14209657/python-call-constructor-of-its-own-instance
[
There is even a global class attribute `__class__`](https://discuss.python.org/t/implicit-accessing-of-class/21032), it's actually the reason why `super().__init__` works (TIL)
```python
class Nok:
  def foo(self):
      return __class__()

nok = Nok()
nok.foo()
>> <__main__.Nok object at 0x103401ab0>